### ISSUE TITLE
NVDA no longer reports changes in ARIA-expanded in Firefox 53.0.3

### ISSUE THREAD

**Issue by pbossley** [nvaccess/nvda#7261]
### Steps to reproduce:
1. Launch Firefox 53.0.3
2. Move focus to an element with ARIA-expanded (example at http://online.osu.edu/ the "toggle search")
3. NVDA properly reports the current state e.g. collapsed or expanded.
4. Toggle the state via space or enter.

### Expected behavior:
NVDA should announce "expanded" or "collapsed" respectively based on the new value.

### Actual behavior:
Nothing is announced.

### System configuration:
NVDA version: 2017.2
NVDA Installed or portable: Installed

Windows version: Windows 10 Enterprise Version	10.0.10240 Build 10240
 ### Other questions:

Does the issue still occur after restarting your PC? Yes

Have you tried any other versions of NVDA? No - but NVDA 2017.1 did not have this behavior.

**Comment by jcsteh** [comment]
I can confirm the issue. However, I just tested this with NVDA 2017.1 and was still able to reproduce the issue. This suggests there has either been a change in Firefox or the site.

This is a duplicate of #5247. Brief explanation (from https://github.com/nvaccess/nvda/issues/6401#issuecomment-249069135):

> Firefox is regenerating the accessible for that button when it expands/collapses. So, instead of firing a stateChange, it fires focus on a "new" accessible. See Mozilla bugs [659886](https://bugzilla.mozilla.org/show_bug.cgi?id=659886) and [1285862](https://bugzilla.mozilla.org/show_bug.cgi?id=1285862).
> 
> If it were a stateChange, NVDA would report it. Instead, NVDA suppresses reporting the focus event because it's at the same position in the document (and not doing this would result in a lot of duplication when moving around).

This really needs to be fixed in Firefox. On the NVDA side, this might be improved by #2039, which will hopefully report focus better in these problematic cases. However, even if we manage that, it will still not report just "expanded"; it will report the entire control, since a new control got focus.

**Comment by pbossley** [comment]
There was a Firefox update around the same time so it is maybe the issue. What I don't understand is that JAWS 18 seems to be able to handle the reporting correctly. I wonder if FS is hacking around the issue. The other strange thing is that NVDA doesn't behave like this when the expanding element is a link or button. In those cases, the change in state is reported as would be expected.The bugs you reference appear to be long standing issues and this seems to be a recent development. I will try to isolate further by trying earlier firefox versions. And if we can lend support to having Mozilla address these issues I'm happy to add my +1 to those.
Thank you,

**Comment by jcsteh** [comment]
Does JAWS say just "collapsed"/"expanded" when you toggle or does it read
the entire control? If it's reading the entire control, it's probably just
treating it as focus on a new control. Ideally, this is how NVDA should
handle this too, but it doesn't yet because of #5825. To put this another
way, the fact that Firefox reports this as a focus change instead of a
state change is a Firefox bug. The fact that NVDA doesn't handle the focus
change is an NVDA bug. Ideally, both should be fixed (and we hope to tackle
the NVDA bug later this year).

**Comment by pbossley** [comment]
Understood. Yes, JAWS reports the entire control +expanded or collapsd. So it sounds like getting to that level is something you have planned for the future. But also there is still a FF bug. Given the state of those, do you think they will ever fix them? The one in particular has been open for 6 years...
Interestingly NVDA reports the control state change in IE but when initially landing on it does not report that it has a state. Wish browsers would convey information in a single way.
Thanks!
